cmake/FindOpenCL.cmake

```diff
@@ -47,18 +47,18 @@ function(_FIND_OPENCL_VERSION)
   set(CMAKE_REQUIRED_QUIET ${OpenCL_FIND_QUIETLY})
 
   CMAKE_PUSH_CHECK_STATE()
-  foreach(VERSION "2_2" "2_1" "2_0" "1_2" "1_1" "1_0")
+  foreach(VERSION "3_0" "2_2" "2_1" "2_0" "1_2" "1_1" "1_0")
     set(CMAKE_REQUIRED_INCLUDES "${OpenCL_INCLUDE_DIR}")
 
     if(APPLE)
       CHECK_SYMBOL_EXISTS(
         CL_VERSION_${VERSION}
-        "${OpenCL_INCLUDE_DIR}/Headers/cl.h"
+        "Headers/cl.h"
```

Comment:
I read the stackoverflow post you shared and it seems that this change merely tricks CMake to make it pick up the header files you want it to pick up because removing the full path to the header files adds ambiguity that resolves in your favor.

Have you tried to unset `OpenCL_INCLUDE_DIR` prior to calling to `find_package(OpenCL)`?

Reply:
As you can see, `OpenCL_INCLUDE_DIR` is already used in this `.cmake file`, line 51 `set(CMAKE_REQUIRED_INCLUDES "${OpenCL_INCLUDE_DIR}")`, but it does not help in case of multiple versions installed.
But in my case I need multiple versions, because NVIDIA do not support latest OpenCL and it ships with old OpenCL header file, that is why in some rare (but very important cases, like to use NVIDIA with OpenCL and oneDNN) it could improve experience ...
I've tried to do multiple thing, including adding defines before and after, but non of it helped, only this little trick helped

Reply:
@densamoilov Is there any other issues with this PR ??
It should work anyway because `set(CMAKE_REQUIRED_INCLUDES "${OpenCL_INCLUDE_DIR}")` specified before `CHECK_SYMBOL_EXISTS`